loki/transformations/array_indexing.py

```diff
@@ -178,6 +182,12 @@ def resolve_vector_notation(routine):
                     # Create new index variable
                     vtype = SymbolAttributes(BasicType.INTEGER)
                     ivar = sym.Variable(name=f'{ivar_basename}_{i}', type=vtype, scope=routine)
+                    # create tuple to test whether an appropriate loop is already available
+                    test_range = (sym.IntLiteral(1), s, 1) if not isinstance(s, sym.RangeIndex)\
+                            else (s.lower, s.upper, 1)
+                    # actually test for it
+                    if test_range in loop_map:
+                        ivar = loop_map[test_range]
```

Comment:
We should test first before creating `ivar` and then overwriting it, because the instantiation creates already an entry in the symbol table:
```suggestion
                    # create tuple to test whether an appropriate loop is already available
                    test_range = (sym.IntLiteral(1), s, 1) if not isinstance(s, sym.RangeIndex)\
                            else (s.lower, s.upper, 1)
                    # actually test for it
                    if test_range in loop_map:
                        ivar = loop_map[test_range]
                    else:
                        vtype = SymbolAttributes(BasicType.INTEGER)
                        ivar = sym.Variable(name=f'{ivar_basename}_{i}', type=vtype, scope=routine)
```